### ISSUE TITLE
AmbiguousActionException... How to force a particular namespace

### ISSUE THREAD

**Issue by Mardoxx** [aspnet/Mvc#6570]
Before we were able to do this with `MapRoute(.... namespaces: ["whatever"])`.

How can you achieve this now?

**Comment by Eilon** [comment]
@Mardoxx - can you share more about your scenario? Maybe there's another way to solve it, such as using Areas?

**Comment by Mardoxx** [comment]
Thanks for the pointer, I sorted it using areas. Just tacked on `[Area("Whatever")]` attribute to my controller and set `area = "Whatever"` in my route defaults.

Long story short, if you're interested... I'm using [imageresizer](https://imageresizing.net/) as a service to process images. One of their plugins `AzureReader2` allows you to go to a specific route in the format of `/{path}/{azureContainer}/{filePath}?{imageResizerQueryString}` and it pulls the image from Azure Storage, processes it and serves it. 

Trouble is it does this using Http Modules which won't work outside of IIS (as far as I'm aware). So I had to create a simple controller which simulates what these http modules do. To avoid cluttering up my projects (and code copy pasting between projects) I separated this into a library. You "install" it with the standard DI sorta pattern. Configure it with `IServiceCollection.AddImageResizer(opts => {/*...*/})` and use it with `IApplicationBuilder.UseImageResizer();`

The way it is mapped is like

    builder.Map(options.Path, app =>
    {
        app.UseMvc(routes =>
        {
            routes.MapRoute("imageresizer", "{*path}",
                new { controller = "AzureImageResizer", action = "AzureStorage" });
        });
    }

When I included this and ran my application, I still had residual controllers left in my project with the same names as those in the library -- raising the AmbiguousActionException. I thought this would be easy enough to fix with the `namespaces[]` filtering which we used to have in MVC-vOld. But this did not work. Removing the controllers made it work fine, but this didn't satisfy me.

The solution was to add `[Area("ImageResizer")]` to my controllers and add `area = "ImageResizer"` to my defaults.
